### PR TITLE
mirror tests, license and new PushSubscribe static instance method

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/examples/java/io/nats/examples/ExampleArgs.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
     public int msgCount = -1;
     public int subCount = -1;
     public String stream = null;
+    public String mirror = null;
```
Comment (scottf) [inline src/examples/java/io/nats/examples/ExampleArgs.java:32]: example args can handle mirror name now

File: src/examples/java/io/nats/examples/ExampleArgs.java [hunk 2]
```diff
@@ -162,77 +166,83 @@ public static Builder builder() {
...
+        private int msgCountDefault = -1;
+        public int subCountDefault = -1;
+        private String streamDefault = null;
+        private String mirrorDefault = null;
+        private String consumerDefault = null;
+        private String durableDefault = null;
+        private String deliverDefault = null;
+        private int pullSizeDefault = 0;
+        private String userDefault = null;
+        private String passDefault = null;
```
Comment (scottf) [inline src/examples/java/io/nats/examples/ExampleArgs.java:181]: made these variable names less ambiguous. when my own code confuses me, I had to change it

File: src/examples/java/io/nats/examples/jetstream/NatsJsMirrorSubUseCases.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import java.util.List;
+
+import static io.nats.examples.jetstream.NatsJsUtils.printFormatted;
+import static io.nats.examples.jetstream.NatsJsUtils.publish;
+
+/**
+ * This example will demonstrate JetStream mirrors subscribing. .
+ */
+public class NatsJsMirrorSubUseCases {
+    static final String usageString =
```
Comment (scottf) [inline src/examples/java/io/nats/examples/jetstream/NatsJsMirrorSubUseCases.java:34]: mirror example

File: src/examples/java/io/nats/examples/jetstream/NatsJsMirrorSubUseCases.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import java.time.Duration;
+import java.util.List;
+
+import static io.nats.examples.jetstream.NatsJsUtils.printFormatted;
+import static io.nats.examples.jetstream.NatsJsUtils.publish;
+
+/**
+ * This example will demonstrate JetStream mirrors subscribing. .
+ */
+public class NatsJsMirrorSubUseCases {
```
Comment (scottf) [inline src/examples/java/io/nats/examples/jetstream/NatsJsMirrorSubUseCases.java:33]: mirror example

File: src/main/java/io/nats/client/PushSubscribeOptions.java [hunk 5]
```diff
@@ -35,6 +35,10 @@ public String getDeliverSubject() {
         return consumerConfig.getDeliverSubject();
     }
 
+    public static PushSubscribeOptions source(String stream) {
+        return new Builder().stream(stream).build();
```
Comment (scottf) [inline src/main/java/io/nats/client/PushSubscribeOptions.java:39]: I landed on using source in this shortcut builder. open for comments.

File: src/main/java/io/nats/client/api/AccountLimits.java [hunk 6]
```diff
@@ -1,3 +1,16 @@
+// Copyright 2020 The NATS Authors
```
Comment (scottf) [inline src/main/java/io/nats/client/api/AccountLimits.java:1]: found this was missing so added it

File: src/main/java/io/nats/client/api/AccountStatistics.java [hunk 7]
```diff
@@ -1,3 +1,16 @@
+// Copyright 2020 The NATS Authors
```
Comment (scottf) [inline src/main/java/io/nats/client/api/AccountStatistics.java:1]: found this was missing so added it

File: src/main/java/io/nats/client/api/Mirror.java [hunk 8]
```diff
@@ -49,7 +49,7 @@ Builder getThis() {
         }
 
         public Mirror build() {
-            return new Mirror(name, startSeq, startTime, filterSubject, external);
+            return new Mirror(sourceName, startSeq, startTime, filterSubject, external);
```
Comment (scottf) [inline src/main/java/io/nats/client/api/Mirror.java:52]: better variable name

File: src/test/java/io/nats/client/impl/JetStreamManagementTests.java [hunk 9]
```diff
@@ -456,238 +455,6 @@ public void testGetAndDeleteMessage() throws Exception {
         });
     }
 
-    @Test
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/JetStreamManagementTests.java:459]: I split the mirror specific tests into their own class

File: src/test/java/io/nats/client/utils/TestBase.java [hunk 10]
```diff
@@ -70,11 +70,11 @@ public static void runInServer(boolean debug, boolean jetstream, InServerTest in
         }
     }
 
-    public static void runAgainstServer(InServerTest inServerTest) throws Exception {
-        runAgainstServer(Options.DEFAULT_URL, inServerTest);
+    public static void runInExternalServer(InServerTest inServerTest) throws Exception {
```
Comment (scottf) [inline src/test/java/io/nats/client/utils/TestBase.java:73]: better name

File: src/main/java/io/nats/client/PushSubscribeOptions.java (hunk 5 above)
Comment (ColinSullivan1) [reply src/main/java/io/nats/client/PushSubscribeOptions.java:45]: I'd suggest coordinating with @wallyqs on naming.  Source works for me.

File: src/test/java/io/nats/client/utils/TestBase.java (hunk 10 above)
Comment (ColinSullivan1) [reply src/test/java/io/nats/client/utils/TestBase.java:73]: `runWithExternalServer` ?

**Review by ColinSullivan1 — APPROVED** [review]
LGTM, please coordinate with @wallyqs  on the name of the property and update if you decide to change it.  Thanks!

File: src/main/java/io/nats/client/PushSubscribeOptions.java (hunk 5 above)
Comment (wallyqs) [reply src/main/java/io/nats/client/PushSubscribeOptions.java:45]: I think the go client we have `nats.BindStream()` to set the stream name for mirrors, would this be similar?

```go
js.SubscribeSync("origin", nats.BindStream("m1"))
```

Comment (scottf) [reply src/main/java/io/nats/client/PushSubscribeOptions.java:45]: changing to `bind` instead of `source`

**Pull request merged**